packages/flutter/test/cupertino/picker_test.dart

```diff
@@ -56,6 +57,61 @@ void main() {
   });
 
   group('scroll', () {
+    testWidgets(
+      'scrolling calls onSelectedItemChanged and triggers haptic feedback',
+      (WidgetTester tester) async {
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+        final List<int> selectedItems = <int>[];
+        final List<MethodCall> systemCalls = <MethodCall>[];
+
+        SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
+          systemCalls.add(methodCall);
+        });
+
+        await tester.pumpWidget(
+          new Directionality(
+            textDirection: TextDirection.ltr,
+            child: new CupertinoPicker(
+              itemExtent: 100.0,
+              onSelectedItemChanged: (int index) { selectedItems.add(index); },
+              children: new List<Widget>.generate(100, (int index) {
+                return new Center(
+                  child: new Container(
+                    width: 400.0,
+                    height: 100.0,
+                    child: new Text(index.toString()),
+                  ),
+                );
+              }),
+            ),
+          ),
+        );
+
+        await tester.drag(find.text('0'), const Offset(0.0, -100.0));
+        expect(selectedItems, <int>[1]);
+        expect(
+          systemCalls.single,
+          isMethodCall(
+            'HapticFeedback.vibrate',
+            arguments: 'HapticFeedbackType.selectionClick',
+          ),
+        );
+
+        await tester.drag(find.text('0'), const Offset(0.0, 100.0));
+        expect(selectedItems, <int>[1, 0]);
+        expect(systemCalls.length, 2);
```

Comment:
Totally optionally, if you like it better: `expect(systemCalls, hasLength(2))`

I did a quick grep of the codebase, and it doesn't look like there's a strong statistical preference for one over the other.

Reply:
Ummm on second thought, the matcher would produce better error messages. Changed.